gatling-jms/src/main/scala/io/gatling/jms/jndi/JmsJndiConnectionFactoryBuilder.scala

```diff
@@ -64,6 +68,10 @@ case class JmsJndiConnectionFactoryBuilder(
       properties.put(Context.SECURITY_CREDENTIALS, credentials.password)
     }
 
+    jndiProperties.foreach { case (k, v) =>
+      properties.put(k, v)
+    }
```

Comment:
Add in imports `import scala.collection.JavaConverters._`

then replace with `properties.putAll(jndiProperties.asJava)`